src/main/kotlin/com/pubnub/api/PubNub.kt

```diff
@@ -86,15 +87,21 @@ import com.pubnub.api.models.consumer.objects.membership.PNChannelDetailsLevel
 import com.pubnub.api.presence.Presence
 import com.pubnub.api.subscribe.Subscribe
 import com.pubnub.api.subscribe.eventengine.configuration.EventEngineConfImpl
+import com.pubnub.api.subscribe.eventengine.effect.RetryPolicy
 import com.pubnub.api.vendor.Base64
 import com.pubnub.api.vendor.Crypto
 import com.pubnub.api.vendor.FileEncryptionUtil.decrypt
 import com.pubnub.api.vendor.FileEncryptionUtil.encrypt
+import com.pubnub.api.workers.SubscribeMessageProcessor
 import java.io.InputStream
 import java.util.Date
 import java.util.UUID
 
-class PubNub internal constructor(val configuration: PNConfiguration, eventEngineConf: EventEngineConf) {
+class PubNub internal constructor(
+    val configuration: PNConfiguration,
+    eventEngineConf: EventEngineConf,
+    retryPolicy: RetryPolicy = configuration.retryPolicy()
```

Comment:
To run contract tests in a timely manner I need to be able to configure the delay between retries. We do not create Subscribe in the tests and due to circular dependency it's not possible to pass it to PubNub instance so it seemed like this is the only option available.

Reply:
My understanding is that in test you can set retryPolicy in configuration and then create PubNub instance containing desired retryPolicy. Is this not enough for test to work properly?

Reply:
It's not enough. We can't change the delay in configuration and without changing the delay these tests would take too much time

Reply:
> My understanding is that in test you can set retryPolicy in configuration 

We don't have such field in configuration

Reply:
So maybe adding `delay` as internal parameter in config would solve this inconvenience.  My point is that adding retryPolicy as parameter to PubNub constructor is something that I would try to avoid.